test/controllers/challenges_controller_test.rb

```diff
@@ -39,6 +39,38 @@ def setup
     assert true, wrong_flag_messages.include?(flash[:notice])
   end
 
+  test 'can not submit flag with no team' do
+    sign_in users(:user_two)
+    put :update, params: {
+        id: challenges(:challenge_one_cat_one),
+        challenge: {
+            submitted_flag: flags(:flag_one).flag
+        }
+    }
+    assert :success
+    assert true, wrong_flag_messages.include?(flash[:notice])
+  end
+
+  test 'can not submit flag with no flag' do
+    sign_in users(:user_one)
+    put :update, params: {
+        id: challenges(:challenge_one_cat_one)
+    }
+    assert :success
+    assert true, wrong_flag_messages.include?(flash[:notice])
+  end
+
+  test 'can not submit flag with no user' do
+    put :update, params: {
+        id: challenges(:challenge_one_cat_one),
```

Comment:
The tabbing seems to be 4 spaces here but 2 for the rest of the document

Reply:
spacing has been removed.  I'm upset rubocop didn't catch that for me